crates/forge/bin/cmd/compiler.rs

```diff
@@ -59,62 +61,208 @@ impl ResolveArgs {
-            }
-
-            println!("{}", serde_json::to_string(&output)?);
-        } else {
-            for (language, sources) in sources {
-                println!("{language}:\n");
+        let mut output: BTreeMap<String, Vec<(Version, Vec<String>)>> = BTreeMap::new();
 
-                for (version, sources) in sources {
-                    if verbosity >= 1 {
-                        println!("{version}:");
-                        for (idx, (path_file, _)) in sources.iter().enumerate() {
-                            let path = path_file
+        for (language, sources) in sources {
+            let mut versions_with_paths: Vec<(Version, Vec<String>)> = sources
+                .iter()
+                .map(|(version, sources)| {
+                    let paths: Vec<String> = sources
+                        .iter()
+                        .map(|(path_file, _)| {
+                            path_file
                                 .strip_prefix(&project.paths.root)
                                 .unwrap_or(path_file)
-                                .to_path_buf();
+                                .to_path_buf()
+                                .display()
+                                .to_string()
+                        })
+                        .collect();
+
+                    (version.clone(), paths)
+                })
+                .collect();
 
-                            if idx == sources.len() - 1 {
-                                println!("└── {}\n", path.display());
-                            } else {
-                                println!("├── {}", path.display());
-                            }
+            versions_with_paths.sort_by(|(v1, _), (v2, _)| Version::cmp(v1, v2));
+
+            output.insert(language.to_string(), versions_with_paths);
+        }
+
+        if json {
+            println!("{}", serde_json::to_string(&output)?);
+            return Ok(());
+        }
+
+        for (language, versions) in &output {
+            if verbosity < 1 {
+                println!("{language}:");
+            } else {
+                println!("{language}:\n");
+            }
+
+            for (version, paths) in versions {
+                if verbosity >= 1 {
+                    println!("{version}:");
+                    for (idx, path) in paths.iter().enumerate() {
+                        if idx == paths.len() - 1 {
+                            println!("└── {path}\n");
+                        } else {
+                            println!("├── {path}");
                         }
-                    } else {
-                        println!("- {language} {version}");
                     }
+                } else {
+                    println!("- {version}");
                 }
             }
+
+            if verbosity < 1 {
+                println!();
+            }
         }
 
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
these tests depend on 1. the latest solc version, 2. the forge-std layout; please fix this and also move them to tests/ directory

Reply:
Thanks, I've moved + updated the tests to use `forgetest!` and pin their pragma's